CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ BUG FIXES:
 * resource/nomad_acl_policy: fixed a bug where the namespace would be incorrectly calculated from a job identity ([#396](https://github.com/hashicorp/terraform-provider-nomad/pull/396))
 * resource/nomad_csi_volume_registration: fixed a bug that cause an import operation to not load all of the volume attributes ([#402](https://github.com/hashicorp/terraform-provider-nomad/pull/402))
 * resource/nomad_job: fixed a bug that could cause jobs to be registered in the incorrect namespace if the `NOMAD_NAMESPACE` environment variable is set ([#386](https://github.com/hashicorp/terraform-provider-nomad/pull/386))
+* resource/nomad_job: fixed a bug that caused a state changes even in case of errors during apply ([#412](https://github.com/hashicorp/terraform-provider-nomad/pull/412))
```

Comment:
super nit:

```suggestion
* resource/nomad_job: fixed a bug that caused a state change even in case of errors during apply ([#412](https://github.com/hashicorp/terraform-provider-nomad/pull/412))
```

or

```suggestion
* resource/nomad_job: fixed a bug that caused state changes even in case of errors during apply ([#412](https://github.com/hashicorp/terraform-provider-nomad/pull/412))
```

Reply:
Ops, I meant the second one. Thanks!